pkg/plugin/admissioncontrollers/webhook/admission_controller.go

```diff
@@ -294,6 +294,41 @@ func (c *admissionController) mutate(req *admissionv1.AdmissionRequest) *admissi
 	return admissionResponseBuilder(uid, true, "", patchBytes)
 }
 
+func (c *admissionController) processWorkload(req *admissionv1.AdmissionRequest) *admissionv1.AdmissionResponse {
+	var uid = string(req.UID)
+	requestKind := req.Kind.Kind
+
+	annotations, supported, err := c.annotationHandler.GetAnnotationsFromRequestKind(requestKind, req)
+	if !supported {
+		// Unknown request kind - pass
+		return admissionResponseBuilder(uid, true, "", nil)
+	}
+	if err != nil {
+		return admissionResponseBuilder(uid, false, err.Error(), nil)
+	}
+
+	if annotation, ok := annotations[userInfoAnnotation]; ok && !c.bypassAuth {
+		userName := req.UserInfo.Username
+		groups := req.UserInfo.Groups
+
+		if allowed := c.annotationHandler.IsAnnotationAllowed(userName, groups); !allowed {
+			errMsg := fmt.Sprintf("user %s with groups [%s] is not allowed to set user annotation", userName,
+				strings.Join(groups, ","))
+			log.Logger().Error("user info validation failed - submitter is not allowed to set user annotation",
+				zap.String("user", userName),
+				zap.Strings("groups", groups))
+			return admissionResponseBuilder(uid, false, errMsg, nil)
+		}
+
+		if err := c.annotationHandler.IsAnnotationValid(annotation); err != nil {
+			log.Logger().Error("invalid user info annotation", zap.Error(err))
+			return admissionResponseBuilder(uid, false, err.Error(), nil)
+		}
+	}
+
+	return admissionResponseBuilder(uid, true, "", nil)
+}
+
```

Comment:
I see a lot of commonalities between these two ways of processing request after username and groups info extraction which can be refactored to a common method and this new common method can be used by these two methods.

Reply:
Done